src/Microsoft.AspNetCore.Authentication.Impl/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -46,8 +45,7 @@ public static IServiceCollection AddAuthentication(this IServiceCollection servi
             return services;
         }
 
-        // REVIEW: rename to just AddScheme?
-        public static IServiceCollection AddSchemeHandler<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions)
+        public static IServiceCollection AddScheme<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions, bool canHandleRequests)
```

Comment:
The new logic will be pushed shortly, I'm just making sure all the tests still work...  

The default routing is basically RemoteOptions single CallbackPath has become many CallbackPaths(to support OIDC's two). canHandleRequests is gone, but there's a new helper extension called AddRemoteScheme, that adds the callback paths to the scheme.

The middleware will call the IAuthenticationSchemeProvider's GetRequestHandlerSchemes with the request path to get back the list of schemes to call ProcessRequest on.  The default implementation just maps the path string to schemes.  

Looking the ASOS, you would just need to add them all as callback events, and should be good to go I think.